sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1543,10 +1543,108 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
     new AnalysisException(s"'$operation' does not support partitioning")
   }
 
-  def mixedRefsInAggFunc(funcStr: String): Throwable = {
-    val msg = "Found an aggregate function in a correlated predicate that has both " +
-      "outer and local references, which is not supported: " + funcStr
-    new AnalysisException(msg)
+  def accessingOuterQueryColumnIsNotAllowed(plan: String, origin: Origin): Throwable = {
+    new AnalysisException(
+      errorClass = "INVALID_SUBQUERY_EXPRESSION",
+      errorSubClass = "ACCESSING_OUTER_QUERY_COLUMN_IS_NOT_ALLOWED",
+      origin = origin,
+      messageParameters = Array(if (SQLConf.get.includePlansInErrors) s": $plan" else ""))
```

Comment:
General comment: Please use Map to guard against ordering errors: Map("plan" -> ....)
(my comment on removing <plan> not withstanding)